### DISCUSSION TITLE
Trying to position text above images

### DISCUSSION THREAD

**Discussion by xtiaan3** [ImageMagick/ImageMagick#5884] [Help]
I am trying to do two things here with no luck so far. I need to position the text above the robot images. The robot group should be south and the text should be north. Also, I'd like to be able to bring all the robots in a little tighter together and over lap then slightly. I have tried playing around with -geometry but can't seem to work it out. Any help would be appreciated. Attached is what Im currently getting 
![639b8ecfd22ff](https://user-images.githubusercontent.com/16021630/207969668-41e2f7ce-a281-48ad-9500-7d66b5f991c3.jpg)

**Comment by xtiaan3** [comment]
Here is the code 

```
$banner = exec("convert {$robot1} -transparent '#{$hex1}' -resize $size -gravity south \
{$robot2} -transparent '#{$hex_bg2}' -resize $size -gravity south -geometry +400+0 \
 {$robot3} -transparent '#{$hex_bg3}' -resize $size -gravity south -geometry +800+0 \
 {$robot4} -transparent '#{$hex_bg4}' -resize $size -gravity south -geometry -400+0 \
 {$robot5} -transparent '#{$hex_bg5}' -resize $size -gravity south -geometry -400+0 \
\( -size 1400x500 xc:none \
-font 'signpainter-2.ttf' -gravity north -pointsize 88 -fill 'white' -annotate +0+0 'Hello World' \
-trim +repage \) \
-background '#{$hex2}' +smush 0 -extent 1400x500 \
{$output}");
```

**Comment by snibgo** [comment]
You read an image, then resize it. Then you read another image and resize both images. Then you read another image and resize all three images. And so on. Why not read all the inputs, and resize them all with a single operation?

Similarly you set a "-geometry". Before you use it, you set it to another value. And then again to another value.

**Comment by xtiaan3** [comment]
Thank you. Im pretty new to Imagemagick and I really don't know what Im doing. Im trying to read through and understand the documentation. Each robot image that I want to place on the banner has to have the background removed `{$robot2} -transparent '#{$hex_bg2}'` so Im not sure if that makes a difference. Would you be able to show me some example code?

**Comment by snibgo** [comment]
You have the same problem making colours transparent. First, you read an image and make one of its colours transparent. Then you read a second image. Then you make a colour in *both* of those images transparent. Then you read a third image and then you make a colour in *all three* images transparent. And so on.

I suggest you use parentheses (like you have done with the 'Hello world' image) to limit processing to just the relevant image.
